src/threadnames.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <threadnames.h>
+#include <tinyformat.h>
+
+#include <boost/thread.hpp>
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h> // For HAVE_SYS_PRCTL_H
+#endif
+
+#if (defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__DragonFly__))
+#include <pthread.h>
+#include <pthread_np.h>
+#endif
+
+#ifdef HAVE_SYS_PRCTL_H
+#include <sys/prctl.h> // For prctl, PR_SET_NAME
+#endif
+
+
+extern int LogPrintStr(const std::string &str);
+
+std::unique_ptr<ThreadNameRegistry> g_thread_name_registry(new ThreadNameRegistry());
+
+namespace {
```

Comment:
Is there somewhere more general or useful I should put the two functions in this namespace?